modules/navier_stokes/src/fvkernels/PCNSFVKT.C

```diff
@@ -0,0 +1,269 @@
+  const auto sup_vel_y_elem = interpolate(*_limiter,
+                                          _sup_vel_y_elem[_qp],
+                                          _sup_vel_y_neighbor[_qp],
+                                          _grad_sup_vel_y_elem[_qp],
+                                          *_face_info,
+                                          /*elem_is_up=*/true);
+  const auto sup_vel_y_neighbor = interpolate(*_limiter,
+                                              _sup_vel_y_neighbor[_qp],
+                                              _sup_vel_y_elem[_qp],
+                                              _grad_sup_vel_y_neighbor[_qp],
+                                              *_face_info,
+                                              /*elem_is_up=*/false);
+  const auto sup_vel_z_elem = interpolate(*_limiter,
+                                          _sup_vel_z_elem[_qp],
+                                          _sup_vel_z_neighbor[_qp],
+                                          _grad_sup_vel_z_elem[_qp],
+                                          *_face_info,
+                                          /*elem_is_up=*/true);
+  const auto sup_vel_z_neighbor = interpolate(*_limiter,
+                                              _sup_vel_z_neighbor[_qp],
+                                              _sup_vel_z_elem[_qp],
+                                              _grad_sup_vel_z_neighbor[_qp],
+                                              *_face_info,
+                                              /*elem_is_up=*/false);
+
+  const auto sup_vel_elem = VectorValue<ADReal>(sup_vel_x_elem, sup_vel_y_elem, sup_vel_z_elem);
+  const auto u_elem = sup_vel_elem / _eps_elem[_qp];
+  const auto rho_elem = _fluid.rho_from_p_T(pressure_elem, T_fluid_elem);
+  const auto specific_volume_elem = 1. / rho_elem;
+  const auto e_elem = _fluid.e_from_T_v(T_fluid_elem, specific_volume_elem);
+  const auto sup_vel_neighbor =
+      VectorValue<ADReal>(sup_vel_x_neighbor, sup_vel_y_neighbor, sup_vel_z_neighbor);
+  const auto u_neighbor = sup_vel_neighbor / _eps_neighbor[_qp];
+  const auto rho_neighbor = _fluid.rho_from_p_T(pressure_neighbor, T_fluid_neighbor);
+  const auto specific_volume_neighbor = 1. / rho_neighbor;
+  const auto e_neighbor = _fluid.e_from_T_v(T_fluid_neighbor, specific_volume_neighbor);
+
+  const auto c_elem = _fluid.c_from_v_e(specific_volume_elem, e_elem);
+  const auto c_neighbor = _fluid.c_from_v_e(specific_volume_neighbor, e_neighbor);
+
+  const auto sup_vel_elem_normal = sup_vel_elem * _face_info->normal();
+  const auto sup_vel_neighbor_normal = sup_vel_neighbor * _face_info->normal();
+  const auto u_elem_normal = u_elem * _face_info->normal();
+  const auto u_neighbor_normal = u_neighbor * _face_info->normal();
+
+  const auto pr = computeAlphaAndOmega(u_elem_normal, u_neighbor_normal, c_elem, c_neighbor);
+  const auto & alpha = pr.first;
+  const auto & omega = pr.second;
+
+  if (_eqn == "mass")
```

Comment:
so you're doing this selection on equation because everything before then is re-used. I think I like this better than having 3 different strong BC objects for each equation. I ll allow it lol. 

If you could specify a list of equations and a list of variables, would be even cooler. I think the latter is not possible right

Reply:
especially with the momentum checks

Reply:
Well definitely not with the momentum checks. You could circumvent this:

> a list of variables, would be even cooler. I think the latter is not possible right

by overriding `computeResidual` but you would need the logic to loop over an arbitrary number of `variable`-esque args

Reply:
Yeah so code wise it s cleaner that way. Performance wise, most likely not limiting